src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3693,6 +3727,73 @@ void Lowering::LowerHWIntrinsicToScalar(GenTreeHWIntrinsic* node)
         LowerNode(cast);
     }
 }
+
+//----------------------------------------------------------------------------------------------
+// Lowering::LowerHWIntrinsicToScalar: Lowers a tree AND(X, ADD(X, -1) to HWIntrinsic::ResetLowestSetBit
+//     Returns the recplacement node if one is created else nullptr indicating no replacement
+//
+// Parameters
+//     tree   -   GentreePtr of binOp
+//
+GenTree* Lowering::LowerAndOpToResetLowestSetBit(GenTree* node)
+{
+    GenTree* op1 = node->gtGetOp1();
+    GenTree* op2 = node->gtGetOp2();
+    if (op1->OperIs(GT_LCL_VAR) && op2->OperIs(GT_ADD) && !comp->lvaGetDesc(op1->AsLclVar())->IsAddressExposed())
+    {
+        GenTree* addOp1 = op2->gtGetOp1();
+        GenTree* addOp2 = op2->gtGetOp2();
+        if (addOp2->IsIntegralConst(-1) && addOp1->OperIs(GT_LCL_VAR) &&
+            addOp1->AsLclVar()->GetLclNum() == op1->AsLclVar()->GetLclNum() &&
+            ((op1->TypeGet() == TYP_LONG && comp->compOpportunisticallyDependsOn(InstructionSet_BMI1_X64)) ||
```

Comment:
```suggestion
            ((op1->TypeIs(TYP_LONG) && comp->compOpportunisticallyDependsOn(InstructionSet_BMI1_X64)) ||
```
(Avoids the need to parenthesize `op1->TypeGet() == TYP_LONG`)

Reply:
Changed.